api/v1beta1/nodemodulesconfig_types.go

```diff
@@ -21,6 +21,7 @@ import (
 )
 
 type ModuleConfig struct {
+	KernelVersion  string `json:"kernelVersion"`
```

Comment:
Why do we need a `KernelVersion` field? Can't we assume that all modules in the `NodeMoudulesConfigSpec` should run on the node's kernel?

Reply:
in order to deal with race conditions between updating the MNC data of the module, and node's rebooting with new kernel version. Lets say node is rebooted with new kernel version, but module-nmc controller has not updated the NMC entree yet: then NMC controller will try to boot the inappropriate version of the kernel module. Haveing kernel version will allow NMC controller to verify that the module data in the NMC is relevant for that specific node's  kernel version

Reply:
How about a `LastUpdated` timestamp instead? we can then compare it to `node.metadata.creationTimestamp`.
I think having the kernel version is confusing if we just try to solve a race condition. WDYT?

Reply:
why is it confusing? and how would LastUpdated help in that case ( we would add it, but for another issue)? To differentiate between node that just rebooted , to node that rebooted with new kernel, only KernelVersion will help

Reply:
> why is it confusing? and how would LastUpdated help in that case ( we would add it, but for another issue)? To differentiate between node that just rebooted , to node that rebooted with new kernel, only KernelVersion will help

It was confusing to me because I was expecting all modules on a node to have the same kernel version of the node of for dealing with race condition I am used to check timestamps and I didn't understand at first why we need the kernel version.

Since we don't have a way to check timestams anyway then using the kernel version is a good option indeed.